src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/IntegerCopyNumberTransitionProbabilityCacheCollection.java

```diff
@@ -0,0 +1,180 @@
+package org.broadinstitute.hellbender.tools.exome.germlinehmm;
+
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.Reader;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This class constructs a collection of integer copy number transition probability caches, allowing
+ * streamlined query of the transition matrix for each contig, sex genotype, and distance.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class IntegerCopyNumberTransitionProbabilityCacheCollection implements Serializable {
+
+    private static final long serialVersionUID = -7708106959687292204L;
+
+    private final Map<SexGenotypeContigPairKey, IntegerCopyNumberTransitionProbabilityCache>
+            sexGenotypeAndContigToTransitionMatrixCacheMap;
+
+    private final boolean padded;
+    private final int maxCopyNumber;
+    private final Set<String> sexGenotypesSet;
+    private final Set<String> contigsSet;
+
+    /**
+     * Public constructor from a {@link IntegerCopyNumberTransitionMatrixCollection}
+     *
+     * @param transitionMatrixCollection collection of transition matrices
+     */
+    public IntegerCopyNumberTransitionProbabilityCacheCollection(
+            @Nonnull final IntegerCopyNumberTransitionMatrixCollection transitionMatrixCollection,
+            final boolean padStates) {
+        sexGenotypeAndContigToTransitionMatrixCacheMap = new HashMap<>();
+        constructCacheMapFromMatrixCollection(padStates ? transitionMatrixCollection.cloneWithPaddedStates()
+                : transitionMatrixCollection);
+        padded = padStates;
+        if (padded) {
+            maxCopyNumber = transitionMatrixCollection.getMaxCopyNumberInCollection();
+        } else {
+            maxCopyNumber = 0;
+        }
+        sexGenotypesSet = Collections.unmodifiableSet(transitionMatrixCollection.getSexGenotypes());
+        contigsSet = Collections.unmodifiableSet(transitionMatrixCollection.getContigs());
+    }
+
+    /**
+     * Public constructor from a transition matrix collection reader
+     *
+     * @param reader
+     * @param parentPath
+     * @param padStates
+     */
+    public IntegerCopyNumberTransitionProbabilityCacheCollection(@Nonnull final Reader reader,
+                                                                 @Nonnull final String parentPath,
+                                                                 final boolean padStates) {
+        final IntegerCopyNumberTransitionMatrixCollection transitionMatrixCollection =
```

Comment:
Agreed; I got rid of the extra constructors.

Regarding the `Reader` business -- whenever we have a method that takes a `File`, we also _need_ to write a similar method that takes a `Reader` if we provision to read from HDFS or a google bucket. Pretty annoying. In this case, though, I think it is excessive. These prior tables are small and can be copied to the master node on runtime using `-- files <...>` argument of `gatk-launch`.